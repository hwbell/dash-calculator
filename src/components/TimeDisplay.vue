<template>
  <div class="time-display-component">
    <div v-if="showTime" class="space-all-row" id="current-time">
      <span class="app-title">
        <strong>Dashboard</strong> 
      </span>
      <span class="time-date">
        {{ timeDate }}, <span class="time-hours">{{ timeHours }}</span>
      </span>
    </div>
  </div>
</template>

<script>
// import ExpressionWindow from './ExpressionWindow';

export default {
  name: "TimeDisplay",
  props: {
    showTime: Boolean,
  },
  created: function() {
    const moment = require("moment");

    this.timeDate = moment().format("MMMM Do YYYY");
    this.timeHours = moment().format("h:mm a");
    setInterval(() => {
      console.log("interval");
      this.timeDate = moment().format("MMMM Do YYYY");
      this.timeHours = moment().format("h:mm a");
    }, 60000);
  },
  // components: {ExpressionWindow},
  data: function() {
    return {
      timeDate: "",
      timeHours: "",
      message: "This is a TimeDisplay component for us to start with!",
    };
  },
  // computed: {
  //   // a computed getter
  //   reversedMessage: function () {
  //     // `this` points to the vm instance
  //     return this.message.split('').reverse().join('')
  //   },
  // },
  // methods: {
  //   greet: function () {
  //     // `this` inside methods point to the Vue instance
  //     this.message = "Nice to meet you!";
  //   }
  // },
};
</script>

<!-- Add "scoped" attribute to limit CSS to this component only -->
<style scoped>
.time-display-component {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}
.app-title {
  font-size: 18pt;
  margin-left: 40px;
  /* color: white; */
}
.time-hours {
  color: rgb(55, 0, 255);
}
#current-time {
  width: 100%;
  font-size: 14pt;
}
</style>
