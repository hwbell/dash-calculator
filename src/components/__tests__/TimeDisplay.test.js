import { mount } from '@vue/test-utils'
import TimeDisplay from '../TimeDisplay.vue'
import Vue from 'vue';

const message = 'This is a TimeDisplay component for us to start with!';

describe('TimeDisplay', () => {

  let wrapper;
  beforeEach(() => {
    wrapper = mount(TimeDisplay)
  })

  test('is a Vue instance', () => {
    expect(wrapper.isVueInstance()).toBeTruthy()
  })

  test('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot()
  })

  test('renders the right components', async () => {
    let messageDiv = wrapper.findAll('.time-display-component');
    expect(messageDiv.length).toBe(1);

    // time needs the showTime boolean to render
    let timeDiv = wrapper.findAll('#current-time');
    expect(wrapper.findAll('#current-time').length).toBe(0);

    // change the boolean
    wrapper.setProps({ showTime: true });
    await Vue.nextTick();
    expect(wrapper.findAll('#current-time').length).toBe(1);


  })

  // test(`assigns the computed value on the component properly`, () => {
  //   let revMessage = message.split('').reverse().join('');
  //   expect(wrapper.vm.reversedMessage).toBe(revMessage);
  // })

  // test('should change the message when the greet method is called', () => {
  //   wrapper.vm.greet();
  //   expect(wrapper.vm.message).toBe('Nice to meet you!');
  // })
})