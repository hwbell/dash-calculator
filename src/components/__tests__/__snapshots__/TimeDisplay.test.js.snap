// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimeDisplay renders correctly 1`] = `
<div
  class="time-display-component"
>
  <!---->
</div>
`;
